src/main/java/mate/academy/spring/service/impl/CustomUserDetailsService.java

```diff
@@ -0,0 +1,33 @@
+package mate.academy.spring.service.impl;
+
+import static org.springframework.security.core.userdetails.User.withUsername;
+
+import mate.academy.spring.model.User;
+import mate.academy.spring.service.UserService;
+import org.springframework.security.core.userdetails.User.UserBuilder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email)
+            throws UsernameNotFoundException {
+        User user = userService.findByEmail(email).orElseThrow(() ->
+                new UsernameNotFoundException("User with email: " + email + "doesn't exist"));
+        UserBuilder builder = withUsername(user.getEmail());
+        builder.password(user.getPassword());
+        builder.authorities(user.getRoles().stream()
```

Comment:
your authorization does not work at the moment because you use .authorities() here
and hasRole() in SecurityConfig

have you tested your code?

Reply:
It works but  now I understood not for all endpoints